desk/app/heap.hoon

```diff
@@ -180,6 +209,40 @@
       (take-groups !<(=action:g q.cage.sign))
     ==
   ==
+++  watch-epic
+  |=  her=ship
+  ^+  cor
+  =/  =wire  /epic
+  =/  =dock  [her dap.bowl]
+  ?:  (~(has by wex.bowl) [wire dock])
+    cor
+  (emit %pass wire %agent [her dap.bowl] %watch /epic)
+::
+++  take-epic
+  |=  =sign:agent:gall
+  ^+  cor
+  ?+    -.sign  cor
+      %kick
+    (watch-epic src.bowl)
+  ::
+      %fact
+    ?.  =(%epic p.cage.sign)
+      ~&  '!!! weird fact on /epic'
+      cor
+    =+  !<(=epic:e q.cage.sign)
+    ?.  =(epic okay)
+      cor
+    ~&  >>  "good news everyone!"
+    %+  roll  ~(tap by stash)
+    |=  [[=flag:g =heap:h] out=_cor]
+    ?>  =(src.bowl p.flag)
+    he-abet:he-sub:(he-abed:he-core:out flag)
+      %watch-ack
+    %.  cor
+    ?~  p.sign  same
+    (slog leaf/"weird watch nack" u.p.sign)
+  ==
+::
```

Comment:
Is it possible for the publisher to skip directly from behind the subscriber to ahead of the subscriber? If so, will this be handled correctly? Say the subscriber is at 5, and the publisher goes from 4 to 6 without us getting a `fact` for 5.

Reply:
The fix for this is announcing the epic on watch paths, before giving the %chat-logs or w.e.